Sources/Auth/Services/Wallet/WalletRequestSubscriber.swift

```diff
@@ -2,48 +2,58 @@ import Combine
 import Foundation
 import WalletConnectUtils
 import JSONRPC
+import WalletConnectKMS
 
 class WalletRequestSubscriber {
     private let networkingInteractor: NetworkInteracting
     private let logger: ConsoleLogging
+    private let kms: KeyManagementServiceProtocol
     private let address: String?
     private var publishers = [AnyCancellable]()
     private let messageFormatter: SIWEMessageFormatting
     var onRequest: ((_ id: RPCID, _ message: String) -> Void)?
 
     init(networkingInteractor: NetworkInteracting,
          logger: ConsoleLogging,
+         kms: KeyManagementServiceProtocol,
          messageFormatter: SIWEMessageFormatting,
          address: String?) {
         self.networkingInteractor = networkingInteractor
         self.logger = logger
+        self.kms = kms
         self.address = address
         self.messageFormatter = messageFormatter
         subscribeForRequest()
     }
 
     private func subscribeForRequest() {
-        guard let address = address else {return}
-        networkingInteractor.requestPublisher.sink { [unowned self] subscriptionPayload in
+        guard let address = address else { return }
+
+        networkingInteractor.requestPublisher.sink { [unowned self] payload in
+
             logger.debug("WalletRequestSubscriber: Received request")
-            guard
-                let requestId = subscriptionPayload.request.id,
-                subscriptionPayload.request.method == "wc_authRequest" else { return }
-
-            do {
-                guard let authRequestParams = try subscriptionPayload.request.params?.get(AuthRequestParams.self) else { return logger.debug("Malformed auth request params")
-                }
-
-                let message = messageFormatter.formatMessage(
-                    from: authRequestParams.payloadParams,
-                    address: address
-                )
-
-                onRequest?(requestId, message)
-            } catch {
-                logger.debug(error)
-            }
+
+            guard let requestId = payload.request.id, payload.request.method == "wc_authRequest"
+            else { return }
+
+            guard let authRequestParams = try? payload.request.params?.get(AuthRequestParams.self)
+            else { return respondError(.malformedRequestParams, topic: payload.topic, requestId: requestId) }
+
+            let message = messageFormatter.formatMessage(from: authRequestParams.payloadParams, address: address)
+
+            onRequest?(requestId, message)
         }.store(in: &publishers)
     }
 
+    private func respondError(_ error: InternalError, topic: String, requestId: RPCID) {
```

Comment:
would it make sense if this method is async?

Reply:
there is no async context. I think it will be the same